src/uptane/ipuptanesecondary.cc

```diff
@@ -0,0 +1,91 @@
+#include "ipuptanesecondary.h"
+
+namespace Uptane {
+
+bool IpUptaneSecondary::sendRecv(std::unique_ptr<SecondaryMessage> mes, std::shared_ptr<SecondaryPacket>& resp,
+                                 std::chrono::milliseconds to) {
+  pending_messages.clear();
```

Comment:
Can two concurrent `sendRecv` run safely? It looks like one call could delete the response of the other before it can receive it.

Reply:
@lbonn Thread safety hasn't been a goal so far, at least I can't imagine where our code could run sendRecv for the same secondary simultaneously. It also doesn't hurt to add a mutex of course.